src/pybind/mgr/telemetry/module.py

```diff
@@ -1027,6 +1062,12 @@ def compile_report(self, channels: Optional[List[str]] = None) -> Dict[str, Any]
                 'total_bytes': df['stats']['total_bytes'],
                 'total_avail_bytes': df['stats']['total_avail_bytes']
             }
+            # basic_pool_usage collection (2/2)
+            if self.is_enabled_collection(Collection.basic_pool_usage):
+                report['usage']['stats_by_class'] = {} # type: ignore
+                for device_class in df['stats_by_class']:
+                    if device_class in ['hdd', 'ssd', 'nvme']:
+                        report['usage']['stats_by_class'][device_class] = df['stats_by_class'][device_class] # type: ignore
```

Comment:
Again, I implemented this in a for loop to account for cases where there is more than one device class. Not sure if this ever happens, but figured better to be safe.